### ISSUE TITLE
Uncaught TypeError: Cannot read property 'values' of undefined when insppecting ObservableArray in react-devtools

### ISSUE THREAD

**Issue by capaj** [mobxjs/mobx#752]
Error occurs when I use react dev tools to inspect an ObservableArray in a prop in my app:

care?redirect=no:8 Uncaught TypeError: Cannot read property 'values' of undefined
    at Array.eval (eval at <anonymous> (http://localhost:3001/app.js?9eeaba4:1162:1), <anonymous>:1905:114)
    at chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/backend.js:7349:54
    at Array.forEach (native)
    at Bridge._inspectResponse (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/backend.js:7345:54)
    at Bridge._handleMessage (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/backend.js:7264:15)
    at listener (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/backend.js:91:10)
    at nrWrapper (http://be.com/care?redirect=no:8:12747)
(anonymous) @ mobx.js?6a62:1905
(anonymous) @ backend.js:7349
_inspectResponse @ backend.js:7345
_handleMessage @ backend.js:7264
listener @ backend.js:91
nrWrapper @ care?redirect=no:8

the problematics function is this: 
```javascript

function createArrayGetter(index) {
    return function () {
        var impl = this.$mobx;
        if (impl && index < impl.values.length) {
            impl.atom.reportObserved();
            return impl.values[index];
        }
        console.warn("[mobx.array] Attempt to read an array index (" + index + ") that is out of bounds (" + impl.values.length + "). Please check length first. Out of bound indices will not be tracked by MobX");
        return undefined;
    };
}

```
Array in question seems like a normal ObservableArray, serializes into:
```
"["1","1000"]"
```
There is a check for `impl` there already so maybe we should have a warning for a case when `impl` is not defined?

**Comment by capaj** [comment]
I removed the `impl.values.length` from the warning, reloaded and using react-dev-tools again, I clicked to inspect the ObservableArray. This time I got 998 warnings and then this error:
```
care?redirect=no:8 Uncaught TypeError: Cannot read property 'getArrayLength' of undefined
    at Array.get (eval at <anonymous> (http://localhost:3001/app.js?9eeaba4:1162:1), <anonymous>:1826:26)
    at chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/backend.js:7349:54
    at Array.forEach (native)
    at Bridge._inspectResponse (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/backend.js:7345:54)
    at Bridge._handleMessage (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/backend.js:7264:15)
    at listener (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/backend.js:91:10)
    at nrWrapper (http://be.com/care?redirect=no:8:12747)
get @ mobx.js?6a62:1826
(anonymous) @ backend.js:7349
_inspectResponse @ backend.js:7345
_handleMessage @ backend.js:7264
listener @ backend.js:91
nrWrapper @ care?redirect=no:8

```

which is in this getter:
```javascript
    get: function () {
        return this.$mobx.getArrayLength();
    },
```

**Comment by mweststrate** [comment]
@capaj just pushed a commit that fixes it I think, could you verify?

**Comment by capaj** [comment]
@mweststrate no, but I wish I had 😁

**Comment by andykog** [comment]
Please see https://github.com/mobxjs/mobx/issues/610
It can't be fixed

**Comment by andykog** [comment]
And more specific: https://github.com/mobxjs/mobx/issues/410

**Comment by mweststrate** [comment]
@andykog sorry, "fixing" is the wrong term indeed, more like: ignore silently instead of failing loudly when inspecting arrays :)

**Comment by andykog** [comment]
@mwestrate, oh, I see, sorry

**Comment by mweststrate** [comment]
Change has been released as 3.0.2. Although not a definite fix, at least the symptoms are less annoying.

**Comment by refaelos** [comment]
@mweststrate I have the same error (the `Cannot read property 'getArrayLength' of undefined` one) and 3.0.2 still doesn't fix it for some reason ...
